### DISCUSSION TITLE
Multiple new feature ideas

### DISCUSSION THREAD

**Discussion by alberto-lanfranco** [jpanther/congo#43] [Ideas]
First of all, I'd like to thank @jpanther for creating what I consider to be one of the best Hugo themes available 🚀

I'm currently in the process of creating my blog using this theme, and here's some ideas for future improvements: 

- support for "last updated" date besides the publishing date. This could be declared in the front matter using `lastmod` for example
- add the ability to tune text size. I find it to be a little bit on the small side
- add the ability to tune text contrast in dark theme. I find it to be a little bit too grey and low contrast.

Keep up the good work 🙂

**Comment by jpanther** [comment]
Hey @alberto-lanfranco, thanks for your kind words!

I appreciate your suggestions. The last updated date is something I personally wanted to find a way to work into the theme. I just haven't found a place to put it yet that keeps in line with the minimal style. Do you have any thoughts on how you'd like to see this?

Some examples I've played with are things like this but I'm not quite convinced:

![Screen Shot 2021-12-03 at 11 57 06](https://user-images.githubusercontent.com/4462786/144526913-75985410-8661-40c4-9a77-5badd4f619e4.JPG)

![Screen Shot 2021-12-03 at 12 00 31](https://user-images.githubusercontent.com/4462786/144526920-3989426d-1248-422e-9f73-5bf033490b0e.JPG)

With regards to the text size tuning, this is something you can already do using custom CSS. The theme uses relative font sizes throughout so you can simply adjust the base font size and everything will scale accordingly. Create `assets/css/custom.css` in your project folder and add the following:

```css
html {
  font-size: 13pt;
}
```

The default font size is 12pt so you can choose a larger value and it should address your issue.

The text contrast is not so easy to tune as you'll need to adjust the Tailwind configuration and rebuild the theme manually. That said, I do have some tweaks to make to the contrast in some places as there are some items that don't provide enough contrast for accessibility purposes. I'll have a look at the dark appearance as part of this although my general intent is for the dark appearance to be a little less vibrant than the light version.

**Comment by alberto-lanfranco** [comment]
Regarding the update date, I would go with your first example. Publishing date and last update date are two piece of information that are very closely related, I would not separate them.

Regarding the text size, thanks for the tip! I suggest you include it in the Docs as an example of custom CSS.

Looking forward to see the evolution of the project!

**Comment by AntoineSoetewey** [comment]
If I may, I would like to add the following feature:

- On mobile, possibility to choose between the current style for the menu (links below each other) or a hamburger menu

Regards,
Antoine

**Comment by alberto-lanfranco** [comment]
Hey @jpanther,

I noticed that the theme now implements `showDateUpdated`, and has a much better contrast in the dark theme.
Thank you 🙂

I would like to suggest other features that could improve the theme:

- support for headings autonumbering ([here's what I mean](https://codingnconcepts.com/hugo/auto-number-headings-hugo/))
- Inspired by [this theme](https://hugo-terminal.vercel.app):
	- option to add summary below post title when showing a list of posts
	- support for article tags in post lists and post page
	- also a curiosity on my part, for the Terminal theme the specification on how to create the RSS feed is located [here](https://github.com/panr/hugo-theme-terminal/blob/aaaadafec9b6a754cb014e0436d5565d2c3c0fea/layouts/_default/rss.xml). I couldn't find the equivalent file in Congo. Am I missing something?

Thanks for the amazing work 💪

**Reply by jpanther** [reply]
Thanks for the suggestions, a couple of these are already implemented...

- Summaries are available on list pages by enabling the `list.showSummary` [theme parameter](https://jpanther.github.io/congo/docs/configuration/#theme-parameters)
- RSS feeds are implemented using the [default Hugo template](https://github.com/gohugoio/hugo/blob/master/tpl/tplimpl/embedded/templates/_default/rss.xml) so there is no need for a specific file in the theme (although you can provide your own to override the default if you wish)

**Reply by alberto-lanfranco** [reply]
good to know! thanks for the tip 🙂
